bin/load_processed_crashes_into_es.py

```diff
@@ -0,0 +1,203 @@
+WORKER_TIMEOUT = 15 * 60
+
+
+def is_in_storage(crash_storage, crash_id):
+    """Is the processed crash in storage."""
+    return crash_storage.exists_object(f"v1/processed_crash/{crash_id}")
+
+
+def get_threechars():
+    """Generate all combinations of 3 hex digits."""
+    chars = "0123456789abcdef"
+    for x in chars:
+        for y in chars:
+            for z in chars:
+                yield x + y + z
+
+
+def check_elasticsearch(supersearch, crash_ids):
+    """Checks Elasticsearch and returns list of missing crash ids.
+
+    Crash ids should all be on the same day.
+
+    """
+    crash_ids = [crash_ids] if isinstance(crash_ids, str) else crash_ids
+    crash_date = date_from_ooid(crash_ids[0])
+
+    # The datestamp in the crashid doesn't match the processed date sometimes especially
+    # when the crash came in at the end of the day.
+    start_date = (crash_date - datetime.timedelta(days=5)).strftime("%Y-%m-%d")
+    end_date = (crash_date + datetime.timedelta(days=5)).strftime("%Y-%m-%d")
+
+    params = {
+        "uuid": crash_ids,
+        "date": [">=%s" % start_date, "<=%s" % end_date],
+        "_columns": ["uuid"],
+        "_facets": [],
+        "_facets_size": 0,
+        "_fields": FIELDS,
+    }
+    search_results = supersearch.get(**params)
+
+    crash_ids_in_es = [hit["uuid"] for hit in search_results["hits"]]
+    return set(crash_ids) - set(crash_ids_in_es)
+
+
+def check_crashids_for_date(firstchars_chunk, date):
+    """Check crash ids for a given subset of all crash id prefixes and date"""
+    crash_source = build_instance_from_settings(settings.CRASH_SOURCE)
+    crash_dest = build_instance_from_settings(settings.CRASH_DESTINATIONS["es"])
+
+    supersearch = SuperSearch(crash_dest)
+
+    in_crash_source = []
+    missing_in_es = []
+
+    for firstchars in firstchars_chunk:
+        # Grab all the crash ids at the given date directory
+        page_iterator = crash_source.list_objects_paginator(
+            prefix=f"v1/raw_crash/{date}/{firstchars}",
+        )
+
+        for page in page_iterator:
+            # NOTE(bdanforth): Keys here look like /v1/raw_crash/DATE/CRASHID
+            crash_ids = [item.split("/")[-1] for item in page]
+
+            if not crash_ids:
+                continue
+
+            # Check crashstorage source first
+            for crash_id in crash_ids:
+                if is_in_storage(crash_source, crash_id):
+                    in_crash_source.append(crash_id)
+                else:
+                    click.echo(
+                        f"Could not find processed crash for raw crash {crash_id}."
+                    )
+
+            # Check Elasticsearch in batches
+            for crash_ids_batch in chunked(in_crash_source, 100):
+                missing_in_es_batch = check_elasticsearch(supersearch, crash_ids_batch)
+                missing_in_es.extend(missing_in_es_batch)
+
+    return list(set(missing_in_es))
```

Comment:
this chunking occurs inside both the `for firstchars in firstchars_chunk` loop and the `for page in page_iterator`, and given that `in_crash_source` is defined outside of both of those, i believe it also needs to occur outside both those loops.

Additionally, checking elasticsearch should be optional: after discussing with willkg a week ago, it's probably faster and/or more desirable not to check in most cases.

```suggestion
    if only_missing_in_es:
        # Check Elasticsearch in batches
        for crash_ids_batch in chunked(in_crash_source, 100):
            missing_in_es_batch = check_elasticsearch(supersearch, crash_ids_batch)
            missing_in_es.extend(missing_in_es_batch)
        return list(set(missing_in_es))
    else:
        return in_crash_source
```

and then `only_missing_in_es` would need to be added as a boolean click option that gets passed through to this method

Reply:
> this chunking occurs inside both the for firstchars in firstchars_chunk loop and the for page in page_iterator, and given that in_crash_source is defined outside of both of those, i believe it also needs to occur outside both those loops.

Ah I see what you're saying -- the `for` loop to check Elasticsearch is doubly nested under `for firstchars in firstchars_chunk`, but actually, that block is to build up my list of `in_crash_source`, and the Elasticsearch check should be done outside of that block once I have the completed list, `in_crash_source`. Right now I'm performing a bunch of checks multiple times for each prefix and each page in each prefix result. I'll fix that.

FWIW, I copied this code from the `verifyprocessed.py` command I [mentioned earlier](https://github.com/mozilla-services/socorro/pull/6609#discussion_r1608861136), though I didn't catch that it [does this](https://github.com/mozilla-services/socorro/blob/00d064a90eaf533fda9a4ea0c3b1dddf10781c78/webapp/crashstats/crashstats/management/commands/verifyprocessed.py#L100-L103). I'll change it there too.

> Additionally, checking elasticsearch should be optional: after discussing with willkg a week ago, it's probably faster and/or more desirable not to check in most cases.

Can you elaborate on why this is likely faster? More desirable? I can add that flag to the CLI as an option.

Reply:
in the specific case of using this during gcp migration, it's faster and more desirable because we know there is nothing in elasticsearch yet, and that also applies more generally in cases where we know most or all of the crashes being loaded are not in es yet.

i also got the impression from @willkg that checking for crash ids in elasticsearch is very slow, potentially so slow that it's faster to just re-upload a crash than to check if it's present, but i'm not clear on that part.